src/register/register-gnome/gnucash-item-edit.c

```diff
@@ -333,6 +332,32 @@ unblock_toggle_signals(GncItemEdit *item_edit)
 }
 
 
+static gboolean
+draw_arrow_cb (GtkWidget *widget, cairo_t *cr, gpointer data)
```

Comment:
Why not reuse the arrow drawing callback you wrote for the rest of gnucash ? That is if you add direction as data instead of item_edit, you have essentially the same interface in both cases. Particularly as you only use item_edit in the function below to extract the arrow direction.